server/svix-server/src/cfg.rs

```diff
@@ -162,6 +168,83 @@ pub struct ConfigurationInner {
     pub internal: InternalConfig,
 }
 
+fn validate_config_complete(
+    config: &ConfigurationInner,
+) -> std::result::Result<(), ValidationError> {
+    match config.cache_type {
+        CacheType::None | CacheType::Memory => {}
+        CacheType::Redis | CacheType::RedisCluster => {
+            if config.redis_dsn.is_none() {
+                return Err(ValidationError {
+                    code: Cow::from("unset redis_dsn"),
+                    message: Some(Cow::from(
+                        "The redis_dsn field must be set if the cache_type is `redis` or `rediscluster`"
+                    )),
+                    params: HashMap::new(),
+                });
+            }
+        }
+    }
+
+    match config.queue_type {
+        QueueType::Memory => {}
+        QueueType::Redis | QueueType::RedisCluster => {
+            if config.redis_dsn.is_none() {
+                return Err(ValidationError {
+                    code: Cow::from("unset redis_dsn"),
```

Comment:
```suggestion
                    code: Cow::from("missing_field"),
```
Probably something like this? I don't think the code should actually be something specific here.